### PR TITLE
[PARMETIS] Remove the symbolic link on Windows

### LINKED ISSUE #7967: [PARMETIS] Issue installing on Windows
The `PARMETIS` artifact https://github.com/JuliaBinaryWrappers/PARMETIS_jll.jl/releases/download/PARMETIS-v4.0.6+1/PARMETIS.v4.0.6.x86_64-w64-mingw32-mpi+microsoftmpi.tar.gz contains the following symlink:
```julia
 "lib/msmpi.dll" => "../bin/msmpi.dll"
```

<details>

```julia
julia> using Tar: Tar

julia> using Pkg: Pkg, PlatformEngines

julia> using Downloads: download

julia> function tar_get_symlinks(tarball_path::AbstractString)
           symlinks_output = Pair{String,String}[]
           Tar.list(`$(PlatformEngines.exe7z()) x $tarball_path -so`) do header::Tar.Header
               if (header.type == :symlink)
                   push!(symlinks_output, header.path=>header.link)
               end
               nothing
           end
           symlinks_output
       end
tar_get_symlinks (generic function with 1 method)

julia> url = "https://github.com/JuliaBinaryWrappers/PARMETIS_jll.jl/releases/download/PARMETIS-v4.0.6+1/PARMETIS.v4.0.6.x86_64-w64-mingw32-mpi+microsoftmpi.tar.gz";

julia> tar_get_symlinks(download(url))
1-element Vector{Pair{String, String}}:
 "lib/msmpi.dll" => "../bin/msmpi.dll"

```

</details>

This prevents it from being installed on Windows in Julia 1.10 due to stricter artifact integrity checking.

This seems very similar to https://github.com/JuliaPackaging/Yggdrasil/pull/7919

CC: @amontoison @vchuravy

### PR DESCRIPTION
**Pull request by amontoison**
close #7967

### CODE REVIEW

**Comment by amontoison** [comment]
@eschnett Do you know why the builds are failing now with OpenMPI?

**Comment by amontoison** [comment]
I can't compile PARMETIS on `x86-64-linux-musl` with `OpenMPI`.
I suspect that OpenMPI is not correctly compiled on this platform.

File: P/PARMETIS/build_tarballs.jl [hunk 1]
```diff
@@ -89,7 +88,9 @@ platforms, platform_dependencies = MPI.augment_platforms(platforms; MPItrampolin
 
 # Avoid platforms where the MPI implementation isn't supported
 # OpenMPI
-platforms = filter(p -> !(p["mpi"] == "openmpi" && arch(p) == "armv6l" && libc(p) == "glibc"), platforms)
+platforms = filter(p -> !(p["mpi"] == "openmpi" && nbits(p) == 32), platforms)
+platforms = filter(p -> !(p["mpi"] == "openmpi" && Sys.isfreebsd(p)), platforms)
+platforms = filter(p -> !(p["mpi"] == "openmpi" && Sys.iswindows(p)), platforms)
```
Comment (giordano) [inline P/PARMETIS/build_tarballs.jl:93]: Compile with OpenMPI 4 instead:
https://github.com/JuliaPackaging/Yggdrasil/blob/d6998f9189d236fcf7ed457875e1fbc4d15dcc52/H/HDF5/build_tarballs.jl#L245

File: P/PARMETIS/build_tarballs.jl [hunk 2]
```diff
@@ -85,7 +84,7 @@ augment_platform_block = """
 """
 
 platforms = supported_platforms()
-platforms, platform_dependencies = MPI.augment_platforms(platforms; MPItrampoline_compat="5.2.1")
+platforms, platform_dependencies = MPI.augment_platforms(platforms; MPItrampoline_compat="5.3.1", OpenMPI_compat="4.1.6, 5")
```
Comment (giordano) [inline P/PARMETIS/build_tarballs.jl:87]: Uhm, if you aren't changing version number of the package you can't change an existing compat bound
```suggestion
platforms, platform_dependencies = MPI.augment_platforms(platforms; MPItrampoline_compat="5.2.1", OpenMPI_compat="4.1.6, 5")
```

Comment (amontoison) [reply P/PARMETIS/build_tarballs.jl]: Thanks Mosè!
Yes, I just wanted to recompile PARMETIS without the symbolic link on Windows so I would like to not bump the version number if it's not needed.

**Pull request merged**